assets/blocks/conditional-content-block/index.js

```diff
@@ -13,6 +13,7 @@ import metadata from './block.json';
 import icon from '../../icons/conditional-content.svg';
 
 export default {
+	...metadata,
 	title: __( 'Conditional Content', 'sensei-lms' ),
 	description: __(
 		'Content inside this block will be shown to the selected subgroup of users.',
```

Comment:
Does anyone know why we have the `title` and `description` outside the block.json?

First I thought it could be for localization purposes, but I've noticed that the block.json title and description could also be localized. Maybe this wasn't the case before?

Reply:
The main intention is the i18n. 
I think it will not work with the script that extract the string between  `__(`, but I don't know how to test it 🤔

Reply:
I have no idea how this works, to be honest. 😄 

Here's what I've noticed so far:
* The `title` and `description` strings from the `block.json` files are generated in the `sensei-lms.pot` file.
![image](https://user-images.githubusercontent.com/1612178/193087378-bb96ee9c-452f-4551-a5ca-262ef6de05d7.png)
* Sensei-Pro has [title/description](https://github.com/Automattic/sensei-pro/blob/67372790f162ae9ebf531c54d0811cbcc7fdb7c6/modules/interactive-blocks/assets/interactive-video/break-point-block/block.json#L3-L4) in their block.json only. Maybe we should ask the other teams if they had any issues with translating `block.json`? cc @Automattic/evergreen @Automattic/zen 
* Gutenberg also has the title/description only in `block.json`. 🤔 

Maybe I should just test it using loco translate and see if it works?

Reply:
@m1r0 Unfortunately this file is ignored by the script that is extracting the `__(` calls.
You can check here for more information https://nexusp2.wordpress.com/2022/03/22/internationalization-in-sensei-lms/

I don't know if on the sensei-pro they have a different solution.

Reply:
Oh, I get it now, thanks! Ok, so we are leaving both texts if that's the case. 👍

Reply:
Folks, as @m1r0 noticed, on Sensei Pro we have the title/description only in the block.json because `wp i18n make-pot` (offered by WP Cli) is able to recognize the titles/descriptions in block.json.

However, to allow the translations to be recognized by WordPress, we need an additional field on `block.json`: `textdomain`, which you can see on [Sensei Pro](https://github.com/Automattic/sensei-pro/blob/67372790f162ae9ebf531c54d0811cbcc7fdb7c6/modules/interactive-blocks/assets/interactive-video/break-point-block/block.json#L26) and on [Gutenberg](https://github.com/WordPress/gutenberg/blob/aff7fc229af794afd94e95fe082e66ca49f44a21/packages/block-library/src/avatar/block.json#L8), but not on the changes proposed here.

After adding that field, we shouldn't need to leave both texts anymore, because WordPress will use the text domain defined in the block to load the translation automatically.

Does it make sense? :)

Reply:
More information here: https://awhitepixel.com/blog/how-to-translate-custom-gutenberg-blocks-with-block-json/ :)

Reply:
By the way, I think the `textdomain` here should  be `sensei-lms`, which is the text domains we're using when translating using `__( ...., 'sensei-lms')`

Reply:
Ehhh...sorry for the spam, but.

> Unfortunately this file is ignored by the script that is extracting the __( calls.

What file are you referring to, @gabrielcaires? The `wp i18n make-pot` should work very well pulling data from `block.json` without any errors. :thinking:

Reply:
@m1ro Nice, thanks! After that change, I think we can remove the title and description from the JS files, no? The translations should work correctly, now. :)

Reply:
I'm not sure anymore, to be honest. 😄

It looks like our `wp i18n make-pot` is pulling the strings correctly from the `block.json` files and wordpress.org is using the [same command](https://meta.trac.wordpress.org/browser/sites/trunk/wordpress.org/public_html/wp-content/plugins/plugin-directory/cli/i18n/class-code-import.php?rev=10984#L52), right?

> Unfortunately this file is ignored by the script that is extracting the __( calls.

@gabrielcaires Didn't we update this? Are we not using the same localization strategy as wordpress.org (`wp i18n make-pot`)? Sorry, but I'm still a bit confused...

Reply:
> It looks like our wp i18n make-pot is pulling the strings correctly from the block.json files

From my tests, yes! That's why I'm not sure what file @gabrielcaires was referring about.

> and wordpress.org is using the [same command](https://meta.trac.wordpress.org/browser/sites/trunk/wordpress.org/public_html/wp-content/plugins/plugin-directory/cli/i18n/class-code-import.php?rev=10984#L52), right?

I'm not sure about this, but...I think it should. :) 

---- 

With that said, if `wp i18n make-pot` is pulling the strings correctly, in theory, it should be working without any issues, and then we can remove the `__(` calls and also the title/description from the JS files. It's worth testing, though (like, remove the title and description for a given block and test if the title/description is still translated when using a language that is not English on WordPress).

Reply:
We can test if only having them in block.json works by building the plugin, running `npm run i18n:build`, and creating some translations. (For that the easiest I think is using the Loco Translate plugin, where you can add a language and change strings in WP-Admin.)

Reply:
Given that these block titles/descriptions are AFAIK already translated, I think we don't even need to create new translations, to be honest. Just changing the WordPress language and pulling the new translations should work alright. :)